src/SignalR.Orleans/OrleansHubLifetimeManager.cs

```diff
@@ -40,38 +30,26 @@ public class OrleansHubLifetimeManager<THub> : HubLifetimeManager<THub>, IDispos
             this._logger = logger;
             this._clusterClient = clusterClient;
 
+            _logger.LogInformation("Initializing: Orleans HubLifetimeManager {hubName} (serverId: {serverId})...", _hubName, _serverId);
             this.SetupStreams().Wait();
+            _logger.LogInformation("Initialized complete: Orleans HubLifetimeManager {hubName} (serverId: {serverId})", _hubName, _serverId);
         }
 
         private async Task SetupStreams()
         {
             this._streamProvider = this._clusterClient.GetStreamProvider(Constants.STREAM_PROVIDER);
             this._serverStream = this._streamProvider.GetStream<ClientMessage>(_serverId, Constants.SERVERS_STREAM);
-            this._allStream = this._streamProvider.GetStream<AllMessage>(Constants.ALL_STREAM_ID, this._hubTypeName);
+            this._allStream = this._streamProvider.GetStream<AllMessage>(Constants.ALL_STREAM_ID, Constants.STREAM_PER_HUBNAME(this._hubName));
 
             var subscribeTasks = new List<Task>();
             var allStreamHandlers = await _allStream.GetAllSubscriptionHandles();
-            if (allStreamHandlers != null && allStreamHandlers.Count > 0)
-            {
-                foreach (var handler in allStreamHandlers)
-                {
-                    subscribeTasks.Add(handler.ResumeAsync((msg, token) => this.ProcessAllMessage(msg)));
-                }
-            }
-            else
+            if (allStreamHandlers != null)

```

Comment:
No, actually this was quite tricky and I had to do some testing to notice this issue.
They are not needed since this class is registered as `singleton` so only one time you need to subscribe. Apart of that when multiple servers are launched of the same `hub type` the first server was getting subscribed, then the second server was being subscribed and it removed the first subscription (somehow). in grains this works differently.

Actually this was causing these two tests failing:
InvokeAllAsync_WithMultipleServers_WritesToAllConnections_Output
InvokeConnectionAsync_OnServer_WithoutConnection_WritesOutputTo_Connection

Reply:
Sorry, I confused myself where this code was. You are correct. It must be subscribed at startup when the type is created and unsubscribed at the dispose time. Thanks